[Test] ProcessConsoleTests - No console complete notification received.

https://download.eclipse.org/eclipse/downloads/drops4/I20220407-0240/testresults/html/org.eclipse.debug.tests_ep424I-unit-win32-java11_win32.win32.x86_64_11.html

org.eclipse.debug.tests.console.ProcessConsoleTests 
testProcessTerminationNotificationWithInputFile 
No console complete notification received.
java.lang.AssertionError: No console complete notification received.at org.junit.Assert.fail(Assert.java:89)at org.junit.Assert.assertTrue(Assert.java:42)at org.eclipse.debug.tests.console.ProcessConsoleTests.processTerminationTest(ProcessConsoleTests.java:285)at org.eclipse.debug.tests.console.ProcessConsoleTests.testProcessTerminationNotificationWithInputFile(ProcessConsoleTests.java:250)at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)at java.base/java.lang.reflect.Method.invoke(Method.java:566)at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)at org.junit.runners.ParentRunner.run(ParentRunner.java:413)at org.junit.runners.Suite.runChild(Suite.java:128)at org.junit.runners.Suite.runChild(Suite.java:27)at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)at org.junit.runners.ParentRunner.run(ParentRunner.java:413)at org.junit.runner.JUnitCore.run(JUnitCore.java:137)at org.junit.runner.JUnitCore.run(JUnitCore.java:115)at org.junit.vintage.engine.execution.RunnerExecutor.execute(RunnerExecutor.java:42)at org.junit.vintage.engine.VintageTestEngine.executeAllChildren(VintageTestEngine.java:80)at org.junit.vintage.engine.VintageTestEngine.execute(VintageTestEngine.java:72)at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)at org.eclipse.test.EclipseTestRunner.runTests(EclipseTestRunner.java:226)at org.eclipse.test.EclipseTestRunner.run(EclipseTestRunner.java:202)at org.eclipse.test.UITestApplication.lambda$0(UITestApplication.java:103)at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)at org.eclipse.swt.widgets.Synchronizer.runAsyncMessages(Synchronizer.java:185)at org.eclipse.swt.widgets.Display.runAsyncMessages(Display.java:4053)at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3658)at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1155)at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1046)at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:644)at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:551)at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:156)at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:152)at org.eclipse.test.UITestApplication.run(UITestApplication.java:48)at org.eclipse.test.UITestApplication.start(UITestApplication.java:118)at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:136)at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:402)at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)at java.base/java.lang.reflect.Method.invoke(Method.java:566)at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)at org.eclipse.equinox.launcher.Main.basicRun(Main.java:596)at org.eclipse.equinox.launcher.Main.run(Main.java:1467)at org.eclipse.equinox.launcher.Main.main(Main.java:1440)at org.eclipse.core.launcher.Main.main(Main.java:44)
-- | -- | -- | --

--- a/debug/org.eclipse.debug.tests/src/org/eclipse/debug/tests/console/ProcessConsoleTests.java
+++ b/debug/org.eclipse.debug.tests/src/org/eclipse/debug/tests/console/ProcessConsoleTests.java
@@ -244,7 +244,7 @@ public class ProcessConsoleTests extends AbstractDebugTest {
 		assertTrue("Failed to prepare input file.", fileCreated);
 		try {
 			ILaunchConfigurationType launchType = DebugPlugin.getDefault().getLaunchManager().getLaunchConfigurationType(LaunchConfigurationTests.ID_TEST_LAUNCH_TYPE);
-			ILaunchConfigurationWorkingCopy launchConfiguration = launchType.newInstance(null, "testProcessTerminationNotificationWithInputFromFile");
+			ILaunchConfigurationWorkingCopy launchConfiguration = launchType.newInstance(null, name.getMethodName());
 			launchConfiguration.setAttribute(IDebugUIConstants.ATTR_CAPTURE_STDIN_FILE, inFile.getAbsolutePath());
 			TestUtil.log(IStatus.INFO, name.getMethodName(), "Process terminates after Console is initialized.");
 			processTerminationTest(launchConfiguration, false);
@@ -267,7 +267,7 @@ public class ProcessConsoleTests extends AbstractDebugTest {
 	public void processTerminationTest(ILaunchConfiguration launchConfig, boolean terminateBeforeConsoleInitialization) throws Exception {
 		final AtomicBoolean terminationSignaled = new AtomicBoolean(false);
 		final Process mockProcess = new MockProcess(null, null, terminateBeforeConsoleInitialization ? 0 : -1);
-		final IProcess process = DebugPlugin.newProcess(new Launch(launchConfig, ILaunchManager.RUN_MODE, null), mockProcess, "testProcessTerminationNotification");
+		final IProcess process = DebugPlugin.newProcess(new Launch(launchConfig, ILaunchManager.RUN_MODE, null), mockProcess, name.getMethodName());
 		@SuppressWarnings("restriction")
 		final org.eclipse.debug.internal.ui.views.console.ProcessConsole console = new org.eclipse.debug.internal.ui.views.console.ProcessConsole(process, new ConsoleColorProvider());
 		console.addPropertyChangeListener(event -> {
@@ -281,8 +281,7 @@ public class ProcessConsoleTests extends AbstractDebugTest {
 			if (mockProcess.isAlive()) {
 				mockProcess.destroy();
 			}
-			TestUtil.waitForJobs(name.getMethodName(), 50, 10000);
-			assertTrue("No console complete notification received.", terminationSignaled.get());
+			waitWhile(__ -> !terminationSignaled.get(), 10_000, __ -> "No console complete notification received.");
 		} finally {
 			consoleManager.removeConsoles(new IConsole[] { console });
 			TestUtil.waitForJobs(name.getMethodName(), 0, 10000);

